src/main/java/bio/terra/service/snapshot/SnapshotDao.java

```diff
@@ -155,18 +155,17 @@ public void lock(UUID snapshotId, String flightId) {
     MapSqlParameterSource params =
         new MapSqlParameterSource().addValue("id", snapshotId).addValue("flightid", flightId);
     int numRowsUpdated = jdbcTemplate.update(sql, params);
-
-    // if no rows were updated, then throw an exception
-    if (numRowsUpdated == 0) {
-      // this method checks if the snapshot exists
-      // if it does not exist, then the method throws a SnapshotNotFoundException
-      // we don't need the result (snapshot summary) here, just the existence check, so ignore the
-      // return value.
-      retrieveSummaryById(snapshotId);
-
+    String exclusiveLock = retrieveSnapshot(snapshotId).getResourceLocks().getExclusive();
+    if (numRowsUpdated == 0 || exclusiveLock == null) {
       throw new SnapshotLockException(
           "Failed to lock the snapshot", LockOperation.LOCK_EXCLUSIVE.getErrorDetails());
     }
+    if (exclusiveLock != flightId) {
+      throw new SnapshotLockException(
+          "Failed to lock the snapshot with the correct lock",
+          LockOperation.LOCK_EXCLUSIVE.getErrorDetails());
+    }
+    return exclusiveLock;
   }
```

Comment:
Drive by change: Since we're allowing locked snapshots to be returned on retrieve, the "retrieveSummaryById" no longer performs a check if the snapshot is locked.